### PR TITLE
1097 stop area view part 3

### PR DESCRIPTION
**Pull request by ahjyrkia**
closes https://github.com/HSLdevcom/jore-map-ui/issues/1097

BE: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/138

### CODE REVIEW

**Comment by Divinare** [comment]
As I went through the list of things to do in this task, found only 3 of them done? More caution needed, no PR iterating.

StopArea window:
? - termid shouldnt be required field
x - stopArea's transitType should always be editable (also when editing existing stop area)
? - Fix stopArea validations
x - stopAreaGroup can only be edited if isNewStopArea = true
?- Show stops that belong to the stop area
?    * On map, center map to the stops at initial page load
?    * On sidebar, clicking each stop centers it on map
? - Created stopArea should be selected when returned to stopForm

StopForm window: 
? - Always show "Luo uusi pysäkkialue" and edit button, when isEditingDisabled is true/false
x - StopForm.tsx, add (i) button to open a specific pysakkialue
? - StopForm.tsx, changing stopArea should change stop.nameFi, stop.nameSw

How stops belonging to stop area were put into stop the form? Isn't it clear that it is below "StopArea window:" subtopic?

Buttons at stopForm are way too big and should be the same size

---

Added WIP prefix. Only tested this task, didn't look at the code yet since many of the things mentioned in TODO list are missing here (and not included in "TODO in next task" list)

**Comment by ahjyrkia** [comment]
Changes done 

New issues for showing stop area's stops on map and automatically selectin stop area:
https://github.com/HSLdevcom/jore-map-ui/issues/1115
https://github.com/HSLdevcom/jore-map-ui/issues/1116

File: src/components/sidebar/nodeView/StopAreaView.tsx [hunk 1]
```diff
@@ -330,6 +344,41 @@ class StopAreaView extends ViewFormBase<IStopAreaViewProps, IStopAreaViewState>
                             </div>
                         )}
                     </div>
+                    <div className={s.flexRow}>
```
Comment (Divinare) [inline src/components/sidebar/nodeView/StopAreaView.tsx]: I think StopAreaView should be put into its own folder, then this whole table should be put into "stopAreaView/StopTable.tsx"

Our convention has been having each sidebar view in ints own folder, I think it has been a good decision

File: src/components/sidebar/nodeView/StopForm.tsx [hunk 2]
```diff
@@ -195,6 +209,14 @@ class StopForm extends ViewFormBase<IStopFormProps, IStopFormState> {
         navigator.goTo(url);
     };
 
+    private redirectToStopArea = (areaId: string | undefined) => {
+        if (!areaId) return;
```
Comment (Divinare) [inline src/components/sidebar/nodeView/StopForm.tsx]: Remove this line. Instead, if no stopArea is selected, ( i ) button should be disbled (button takes disabled?: boolean as props)

File: src/components/sidebar/nodeView/stopAreaView.scss [hunk 3]
```diff
@@ -17,4 +17,42 @@
...
+            }
+            .columnHeader:first-of-type {
+                padding-left: 0px;
+            }
+
+            .stopHeaderRowHighlight {
+                background-color: $lightblue;
+            }
+        }
+        .editStopHeaderButton {
```
Comment (Divinare) [inline src/components/sidebar/nodeView/stopAreaView.scss]: Is this used?

File: src/components/sidebar/nodeView/stopAreaView.scss [hunk 4]
```diff
@@ -17,4 +17,42 @@
...
+            .columnHeader {
+                white-space: nowrap;
+                text-align: start;
+                padding-left: 10px;
+            }
+            .columnHeader:first-of-type {
+                padding-left: 0px;
+            }
+
+            .stopHeaderRowHighlight {
```
Comment (Divinare) [inline src/components/sidebar/nodeView/stopAreaView.scss]: ?

File: src/models/validationModels/stopAreaValidationModel.ts [hunk 5]
```diff
@@ -10,8 +10,8 @@ const stopAreaValidationModel: IStopAreaValidationModel = {
     transitType: 'required|min:1|max:1|string',
     nameFi: nameRule,
     nameSw: nameRule,
-    terminalAreaId: 'required|min:1|max:10|string',
-    modifiedBy: '',
+    terminalAreaId: 'min:1|max:10|string',
+    modifiedBy: 'min:1|max:20|string',
```
Comment (Divinare) [inline src/models/validationModels/stopAreaValidationModel.ts]: No validation needed for this

File: src/components/sidebar/nodeView/StopAreaView.tsx [hunk 6]
```diff
@@ -174,13 +180,8 @@ class StopAreaView extends ViewFormBase<IStopAreaViewProps, IStopAreaViewState>
...
             this.props.errorStore!.addError(`Tallennus epäonnistui`, e);
         }
-
-        if (this.props.isNewStopArea) {
-            this.navigateToNewStopArea();
-            return;
-        }
-        this.setState({ isLoading: false });
         this.props.stopAreaStore!.setIsEditingDisabled(true);
+        navigator.goBack();
```
Comment (Divinare) [inline src/components/sidebar/nodeView/StopAreaView.tsx]: This should only be done if the save was successful.

Comment (Divinare) [reply src/components/sidebar/nodeView/StopAreaView.tsx]: However, im not even sure if i like it going back. What is the user wants to keep editing after save? Maybe let user click <- instead? Not sure...

Comment (ahjyrkia) [reply src/components/sidebar/nodeView/StopAreaView.tsx]: There is not much to edit so I doubt the user wouldnt do all the changes at once. Though in other views saving something never redirects so I agree that stopAreaView shouldn't be exception to this.

**Comment by ahjyrkia** [comment]
Requested changes done

**Comment by Divinare** [comment]
Review fixes done, added a refactor card https://github.com/HSLdevcom/jore-map-ui/issues/1125

LGTM now

**Pull request merged**